### PR TITLE
gh-91181: drop support for bytes on sys.path

### PR DESCRIPTION
**Pull request by graingert**
Support for bytes broke sometime between Python 3.2 and 3.6 and has been broken ever since. Trying to bring back supports is surprisingly difficult in the face of `-b` and checking for keys in `sys.path_importer_cache`. Since the support was broken for so long, trying to overcome the difficulty of bringing back the support has been deemed not worth it.

* Issue: gh-91181

### CODE REVIEW

File: Doc/library/sys.rst [hunk 1]
```diff
@@ -1125,7 +1125,7 @@ always available.
    the entries inserted as a result of :envvar:`PYTHONPATH`.
 
    A program is free to modify this list for its own purposes.  Only strings
-   and bytes should be added to :data:`sys.path`; all other data types are
+   and should be added to :data:`sys.path`; all other data types are
```
Comment (eryksun) [inline Doc/library/sys.rst]:
```suggestion
   should be added to :data:`sys.path`; all other data types are
```
The edit missed "and" from "strings and bytes".

**Review by MaxwellDupre — APPROVED** [review]
Looks ok.

**Review by jaraco — APPROVED** [review]
Not an authoritative approval, but this approach is my preference.

**Comment by brettcannon** [comment]
Closing in favour of #31897 .

**Review by brettcannon — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2022-03-16-14-24-14.bpo-47025.qtT3CE.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+drop support for bytes on sys.path
```
Comment (brettcannon) [inline Misc/NEWS.d/next/Library/2022-03-16-14-24-14.bpo-47025.qtT3CE.rst]:
```suggestion
Drop support for :class:`bytes` on :attr:`sys.path`.
```

**Review by brettcannon — APPROVED** [review]

**Pull request merged**